packages/compass-workspaces/src/stores/workspaces.ts

```diff
@@ -711,9 +728,22 @@ type CloseTabAction = { type: WorkspacesActions.CloseTab; atIndex: number };
 
 export const closeTab = (
   atIndex: number
-): WorkspacesThunkAction<void, CloseTabAction> => {
-  return (dispatch, getState) => {
+): WorkspacesThunkAction<Promise<void>, CloseTabAction> => {
+  return async (dispatch, getState) => {
     const tab = getState().tabs[atIndex];
+    const { canClose } = resolveTabCloseState(tab);
+    if (!canClose) {
+      const confirmClose = await showConfirmation({
+        title: 'Are you sure you want to close the tab?',
+        description:
+          'The content of this tab was modified. You will loose you changes if you close it',
```

Comment:
[np] Small typo here -- could we change the description to read:
"The content of this tab has been modified. You will lose you changes if you close it."

Reply:
there is also "you changes" -> "your changes" :)

Reply:
Thanks, updated!